Yatima/Cli/PipeCmd.lean

```diff
@@ -1,7 +1,73 @@
 import Cli
+import Yatima.Cli.CompileCmd 
+import Yatima.Transpiler.Transpiler
+import Lean.Util.Path
 
--- TODO
+open System Yatima.Compiler Yatima.Transpiler in
+def pipeRun (p : Cli.Parsed) : IO UInt32 := do
+  match ← getToolchain with
+  | .error msg => IO.eprintln msg; return 1
+  | .ok toolchain =>
+    if toolchain != Lean.versionString then
+      IO.eprintln
+        s!"Expected toolchain '{Lean.versionString}' but got '{toolchain}'"
+      return 1
+  let log := p.hasFlag "log"
+  let eraseTypes := p.hasFlag "no-erase-types"
+  match p.variableArgsAs? String with
+  | some ⟨args⟩ =>
+    if !args.isEmpty then
+      setLibsPaths
+      let mut stt : CompileState := default
+      let mut errMsg : Option String := none
+      let mut cronos := Cronos.new
+      for arg in args do
+        for filePath in ← getLeanFilePathsList ⟨arg⟩ do
+          let filePathStr := filePath.toString
+          cronos ← cronos.clock filePathStr
+          match ← compile filePath log stt with
+          | .ok stt' => match stt.union stt' with
+            | .ok stt' =>
+              stt := stt'
+              cronos ← cronos.clock filePathStr
+            | .error msg => errMsg := some msg; break
+          | .error msg => errMsg := some (toString msg); break
+        if errMsg.isSome then break
+      match errMsg with
+      | some msg =>
+        IO.eprintln msg
+        return 1
+      | none => pure ()
+      match ← transpile stt with
+      | .error msg => IO.eprintln msg
+      | .ok out => 
+        let path ← IO.currentDir
+        let fname : FilePath := path/ s!"output" |>.withExtension "lurk"
```

Comment:
What if we require a `--output` (shortcut: `-o`) from the user where they'd specify the desired file to be written with Lurk code?